### PR TITLE
add option to rerun kubecheck plan (GitHub Only)

### LINKED ISSUE #105: Add ability to force kubechecks to re-run on an MR without new changes
We should add a command like `kubechecks again` to force kubechecks to run again. This is useful in certain edge cases when external dependencies change.

### PR DESCRIPTION
**Pull request by KyriosGN0**
Fixes #105

### CODE REVIEW

**Comment by Greyeye** [comment]
Hi @KyriosGN0, thanks for the PR.
would you mind adding a test case for the ParseHook()?

**Comment by KyriosGN0** [comment]
hi @Greyeye, is there some example on how to do it, im not sure if i should/can mock the entire HTTP request that a webhook sends, would it be possible to have some help?
i did test it with tilt and it works.

**Comment by Greyeye** [comment]
@KyriosGN0 
i've pushed [PR](https://github.com/KyriosGN0/kubechecks/pull/1) to your branch.
it contains a few fixes and unit tests. 
also include changes to GitLab support.

**Comment by KyriosGN0** [comment]
@grey, merged your PR, thx a lot

**Comment by KyriosGN0** [comment]
@Greyeye can you rerun the CI ? i don't see anything weird in the ci logs

**Comment by Greyeye** [comment]
CI is trying to push the docker image to the org repo, and it's failing.
think we need to filter the run when it is executed from forked repo.

@KyriosGN0 can you change

 .github/workflows/on_pull_request.yaml 
 .github/workflows/on_pull_request_closed.yaml 

and add github.repository == 'zapier/kubechecks' under `runs-on`

```
    runs-on: ubuntu-latest
    if: github.repository == 'zapier/kubechecks'
```

**Comment by KyriosGN0** [comment]
@Greyeye added

**Comment by KyriosGN0** [comment]
@Greyeye looks like it failed still

**Comment by KyriosGN0** [comment]
@Greyeye any update ?

File: pkg/vcs/github_client/client.go [hunk 1]
```diff
@@ -143,6 +144,20 @@ func (c *Client) ParseHook(r *http.Request, request []byte) (vcs.PullRequest, er
 			log.Info().Str("action", p.GetAction()).Msg("ignoring Github pull request event due to non commit based action")
 			return nilPr, vcs.ErrInvalidType
 		}
+	case *github.IssueCommentEvent:
+		switch p.GetAction() {
+		case "created":
+			if strings.ToLower(p.Comment.GetBody()) == "kubechecks again" {
```
Comment (djeebus) [inline pkg/vcs/github_client/client.go:150]: Is this the right phrasing? A lot of similar tools would do something like `/kubechecks retry` or something like that. Alternately, we could make it configurable. Still need a good default, but that removes some pressure off of getting it perfectly right.

**Review by djeebus — APPROVED** [review]

**Comment by djeebus** [comment]
Looks great! I think the wording could use some work though; check out my comment?

File: .github/workflows/on_pull-request_closed.yaml [hunk 2]
```diff
@@ -12,6 +12,7 @@ env:
 jobs:
   remove-temp-image:
     runs-on: ubuntu-22.04
+    if: github.repository == 'zapier/kubechecks'
```
Comment (djeebus) [inline .github/workflows/on_pull-request_closed.yaml:15]: I think we just need to add a `continue-on-error: true` here, and the next place

File: pkg/vcs/github_client/client.go (hunk 1 above)
Comment (KyriosGN0) [reply pkg/vcs/github_client/client.go:150]: honestly im coming from atlantis where do `atlantis apply/run apply` but i agree that making it configureable is a good suggestion, i'll update the PR this week and tag you

**Comment by KyriosGN0** [comment]
@djeebus added it

**Comment by Greyeye** [comment]
@KyriosGN0 doc lint and ci are failing, would you be able to check these please.

**Comment by KyriosGN0** [comment]
@Greyeye should be fixed

**Comment by KyriosGN0** [comment]
@Greyeye it seems like same issue, can this PR be merged or is this blocking ? (because its not even something in the code itself)

**Comment by KyriosGN0** [comment]
@Greyeye any updates?

File: pkg/vcs/github_client/client.go [hunk 3]
```diff
@@ -156,15 +171,62 @@ func (c *Client) buildRepoFromEvent(event *github.PullRequestEvent) vcs.PullRequ
...
+	}
+	pathSegments := strings.Split(prURL.Path, "/")
+	if len(pathSegments) < 5 {
+		// invalid path, return nilPr
+		log.Error().Msgf("invalid pr url path: %s", prURL.Path)
+		return nilPr
+	}
+	owner := pathSegments[2]
+	repo := pathSegments[3]
+	prNumber, err := strconv.Atoi(pathSegments[4])
```
Comment (djeebus) [inline pkg/vcs/github_client/client.go:206]: Could we get rid of the string parsing? 

```suggestion
	owner := comment.GetIssue().GetRepository().GetOwner().GetName()
	repo := comment.GetIssue().GetRepository().GetName()
	prNumber := comment.GetIssue().GetNumber()
```

**Comment by djeebus** [comment]
Don't worry about the failing push, it should be (might be?) sorted out if you pull `main`, which you'll need to do to resolve the merge conflicts anyway. I added one comment about the string parsing, but other wise looks good! Mind sorting that out, and we can get this merged?

File: pkg/vcs/github_client/client.go (hunk 3 above)
Comment (KyriosGN0) [reply pkg/vcs/github_client/client.go:206]: sounds ok to me

**Comment by KyriosGN0** [comment]
@djeebus fixed conflicts and commited your suggestion

**Comment by KyriosGN0** [comment]
@djeebus fixed test failures, the change which you proposed caused test to fails as those methods don't error but rather return 0

**Comment by djeebus** [comment]
Thanks! It looks like those failed tests uncovered a broken test, and a legit bug. I've fixed them here, if you want to grab them: https://github.com/KyriosGN0/kubechecks/compare/replan...djeebus:kubechecks:fix-replan-code-tests

**Comment by KyriosGN0** [comment]
@djeebus took your suggestion and also add case for repo/owner is empty string

**Comment by KyriosGN0** [comment]
@djeebus any updates here ?

**Comment by KyriosGN0** [comment]
hey @djeebus @Greyeye any possible news about merging this? will really love to have this feature

**Pull request merged**